pkg/kubectl/run.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/golang/glog"
+)
+
+type BasicReplicationController struct{}
+
+func (BasicReplicationController) ParamNames() []GeneratorParam {
+	return []GeneratorParam{
+		{"labels", false},
+		{"name", true},
+		{"replicas", true},
+		{"image", true},
+		{"command", false},
+	}
+}
+
+func (BasicReplicationController) Generate(params map[string]string) (runtime.Object, error) {
+	labelString, found := params["labels"]
+	var labels map[string]string
+	if found && len(labelString) > 0 {
+		labels = ParseLabels(labelString)
+	} else {
+		labels = map[string]string{
+			"name": params["name"],
```

Comment:
I think creating a label from the name by default is useful, esp. once we support uniquified names, bulk creation, and cloning. However, one issue with calling it `name` is that it won't precisely match the object names in those cases, nor of the pods created by the replication controller. This is why I lean towards something less generic, such as `group`.

Ideally, all labels we add would be namespaced, so as not to conflict with user-provided labels -- we use `kubectl.kubernetes.io/` elsewhere. I understand that the usability argument against that, though. One compromise would be to use the un-namespaced key if the user doesn't specify any labels and the fully qualified, namespaced key if they do.